### PR TITLE
sensors: Driver for INA226

### PR DESCRIPTION
**Pull request by mayankmahajan-nxp**
INA226 - Bidirectional Current and Power Monitor w/ I2C
Boards Tested: mr_canhubk3

### CODE REVIEW

**Comment by mayankmahajan-nxp** [comment]
Hi @MaureenHelm, could you please check this PR. Thank you.

**Comment by mayankmahajan-nxp** [comment]
Hi @MaureenHelm, requesting a review on this PR. Kindly check at your convenience.

File: samples/sensor/ina226/README.rst
Comment (kartben) [inline samples/sensor/ina226/README.rst:1]: please drop this sample entirely -- it is not exercising any API that would be specific to this sensor so is not adding value compared to e.g. sensor_shell sample. 
If you'd like to provide an example of how the binding can be used, you may add the devicetree snippet you have in your mr_canhubk3.overlay to the description field of the binding, as an example.

File: drivers/sensor/ina226/ina226.h
Comment (kartben) [inline drivers/sensor/ina226/ina226.h:28]: copy paste issue? prefix should probably be `INA226_`

**Review by kartben — CHANGES_REQUESTED** [review]

File: samples/sensor/ina226/README.rst
Comment (dleach02) [reply samples/sensor/ina226/README.rst:1]: @kartben is there a sample already in the tree where the author could add the board overlay to show it being tested?

**Comment by mayankmahajan-nxp** [comment]
Hi @kartben, thank you for the review. I have addressed the comments. Kindly check at your convenience.

File: samples/sensor/ina226/README.rst
Comment (kartben) [reply samples/sensor/ina226/README.rst:1]: Like I said, `sensor_shell` sample should be enough for folks to exercise the sensor, since it only uses standard channels and API.
The overlay doesn't belong to code samples since AFAICT the mr_canhub overlay that was previously provided corresponded to a bespoke wiring (i.e. I don't think the mr_canhub comes with an INA226, please correct me if I'm wrong) so it is not really useful for folks. Also hence why I suggested to simply provide a "sample usage" in the description of the binding itself. Hope this helps.

File: drivers/sensor/ina226/ina226.c
Comment (MaureenHelm) [inline drivers/sensor/ina226/ina226.c:230]: Use `i2c_is_ready_dt`

File: dts/bindings/sensor/ti,ina226.yaml
Comment (MaureenHelm) [inline dts/bindings/sensor/ti,ina226.yaml:34]: Is this the power-on reset value? Please explain in the description

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add this sensor to `tests/drivers/build_all/sensor`

**Comment by mayankmahajan-nxp** [comment]
> Please add this sensor to `tests/drivers/build_all/sensor`

Hi @MaureenHelm, thank you for the review. I have addressed the comments. Kindly check at your convenience.

**Comment by dleach02** [comment]
@mayankmahajan-nxp, please clean up the merge conflicts.

@MaureenHelm, @kartben, has the author addressed your review comments?

File: dts/bindings/sensor/ti,ina226.yaml
Comment (MaureenHelm) [inline dts/bindings/sensor/ti,ina226.yaml:55]: Why do we need this property at all then? We shouldn't have redundant properties

**Review by MaureenHelm — COMMENTED** [review]
Needs to be rebased

File: dts/bindings/sensor/ti,ina226.yaml
Comment (mayankmahajan-nxp) [reply dts/bindings/sensor/ti,ina226.yaml]: Yes that's correct. I have removed this property in the latest push. Sorry I didn't see it before.

**Comment by mayankmahajan-nxp** [comment]
Hi @dleach02 and @MaureenHelm, thank you for the review. I have resolved the merge conflicts and review comments. Kindly check at your convenience.

**Review by MaureenHelm — DISMISSED** [review]

**Comment by mayankmahajan-nxp** [comment]
Hi @kartben, requesting your review on this PR. Kindly check at your convenience. Thank you.

File: dts/bindings/sensor/ti,ina226.yaml
Comment (decsny) [inline dts/bindings/sensor/ti,ina226.yaml:76]: this description does not make clear what the value means

Comment (mayankmahajan-nxp) [reply dts/bindings/sensor/ti,ina226.yaml]: Hi @decsny, I have addressed this in the latest push. Kindly check if the description is clear now. If not then please let me know. Thank you.

**Comment by mayankmahajan-nxp** [comment]
Hi @decsny, thank you for the review. I have addressed the comments. Kindly check at your convenience.
Hi @MaureenHelm, your approval was dismissed due to force push for the latest review. Kindly check. Thank you.

**Comment by decsny** [comment]
added dev review label because kartben has not return with a block after some weeks

**Comment by kartben** [comment]
> added dev review label because kartben has not return with a block after some weeks

I missed the notifications due to travelling - sorry. FWIW this kind of stale review is OK to dismiss without going to dev review, as it was clear my comments had been addressed (and thanks for that!)
I just realized I had another comment which I just posted, hopefully it can be addressed but happy if this is done in a follow-up commit if you don't want to lose the existing approvals

**Comment by mayankmahajan-nxp** [comment]
Hi @kartben, thank you for the review. I have addressed the comments. Kindly check at your convenience. Thank you.
@MaureenHelm, @decsny Kindly check if you could approve this PR. I have addressed your comments. Thank you.

File: dts/bindings/sensor/ti,ina226.yaml
Comment (kartben) [inline dts/bindings/sensor/ti,ina226.yaml:7]: missed this small one before - should there be another round of updates: 
```suggestion
    DeviceTree as it provides macros that can be used for initializing the
```

**Review by kartben — DISMISSED** [review]
@mayankmahajan-nxp  thanks!
Sorry if I wasn't clear, but keeping the binding sample was fine, I was only suggesting to have it focus on the "useful" bits. But this works with dropping it altogether too :)

**Review by MaureenHelm — DISMISSED** [review]

**Review by decsny — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]

**Review by decsny — DISMISSED** [review]

**Comment by mayankmahajan-nxp** [comment]
Hi @aescolar, I have rebased. Kindly check at your convenience. Thank you.

**Comment by mayankmahajan-nxp** [comment]
Hi, pushed again just now because a rebase was needed. @aescolar please check at your convenience. Thank you.
@MaureenHelm, @kartben, @decsny your approvals were dismissed due to force push. Kindly check at your convenience.

File: drivers/sensor/ti/ina226/ina226.h
Comment (teburd) [inline drivers/sensor/ti/ina226/ina226.h:16]: There's no need to have a header as this is all local to the ina226.c compilation unit

Comment (mayankmahajan-nxp) [reply drivers/sensor/ti/ina226/ina226.h:16]: Ok @teburd. I could change it. But other INAXXX drivers (such as INA219) include a separate header. Kindly suggest how we should proceed.

**Comment by mayankmahajan-nxp** [comment]
Hi @teburd, thank you for the review. I have addressed your comments. Kindly check at your convenience.

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Pull request merged**
